libavfilter/dnn/dnn_backend_tf.c

```diff
@@ -127,8 +127,9 @@ static DNNReturnType get_input_tf(void *model, DNNData *input, const char *input
     }
     TF_DeleteStatus(status);
 
-    // currently only NHWC is supported
-    av_assert0(dims[0] == 1);
+    //currently only NHWC is supported
+    //some case is = -1
```

Comment:
A few questions:

- Could it be a problem that we have this assertion commented out if the tensor shape is not NHWC?
- Is it the responsibility of the user to provide a Tensorflow model that only uses NHWC (which appears to be the Tensorflow default?)
- When is `dims[0] == 1`?

Reply:
I see so if there happen to be no images in the batch the assertion would trigger a failure, but we don't want to fail immediately in that case.

What is the expected behavior for a batch with no images? Will the rest of the logic handle that case gracefully?

Reply:
In fact, the image exists even if the batch size is 0 or none. This is our case. 
I found the answer [here ](https://github.com/FFmpeg/FFmpeg/blob/dcb285d2b7d126c80fad6079400a904e813442c1/libavfilter/dnn/dnn_backend_tf.c#L157), they allow dims[0] == -1
I will change it as follow.
// currently only NHWC is supported
av_assert0(dims[0] == 1 || dims[0] == -1);